.github/workflows/template.bicep.cleanup.yml

```diff
@@ -0,0 +1,86 @@
+name: "Template-Bicep-Cleanup"
+
+on:
+  workflow_call:
+    inputs:
+      pullRequestNumber:
+        type: string
+        required: true
+      keyVaultName:
+        type: string
+        required: true
+    secrets:
+      azureEnviromentCredentials:
+        required: true
+
+jobs:
+  GetEnvironments:
+    name: GetEnvironments
+    uses: ./.github/workflows/template.queryevents.yml
+    with:
+      pipelineName: PR
+      eventName: NewEnvironment
+      eventGroupId: PR-${{ inputs.pullRequestNumber }}
+      keyVaultName: ${{ inputs.keyVaultName }}
+    secrets:
+      azureEnviromentCredentials: ${{ secrets.azureEnviromentCredentials }}
+
+  Destroy:
+    name: Destroy
+    needs: [GetEnvironments]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Toolset
+        run: |
+          ./setup-azcli.sh
+          source ~/.bashrc
+        working-directory: scripts/orchestrators
+
+      - uses: Azure/login@v1
+        name: Run Azure Login
+        with:
+          creds: "${{ secrets.azureEnviromentCredentials }}"
+          environment: "AzureCloud"
+
+      - name: Get KeyVault Secret(s)
+        id: LoadSecrets
+        env:
+          SECRETS: "clientId, clientSecret, subscriptionId, tenantId"
+          KEY_VAULT_NAME: ${{ inputs.keyVaultName }}
+        run: ./github.loadsecrets.sh
+        working-directory: scripts/orchestrators
+
+      - name: Run Bicep Destroy
+        env:
+          WORKSPACE_PATH: ${{ github.workspace }}
+          RUN_ID: ${{ github.run_id }}
+          NEW_ENVIRONMENT_EVENTS_JSON: ${{ needs.GetEnvironments.outputs.events }}
+        run: |
+          export ARM_TENANT_ID=${{ steps.LoadSecrets.outputs.tenantId }}
```

Comment:
can we set the ARM variables as environment variables in the task env section.

Reply:
As discussed, this requires a broader change in unrelated files. Will create an issue to handle this later.